ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java

```diff
@@ -638,4 +640,18 @@ default void prepareAlterTableEnvironmentContext(AbstractAlterTableDesc alterTab
   default Boolean hasAppendsOnly(org.apache.hadoop.hive.ql.metadata.Table hmsTable, SnapshotContext since) {
     return null;
   }
+
+  /**
+   * Checks if storage handler supports Show Partitions and returns a list of partitions
+   * @param context
+   * @param tbl
+   * @return List of partitions
+   * @throws UnsupportedOperationException
+   * @throws HiveException
+   */
+  default List<String> showPartitions(DDLOperationContext context,
+      org.apache.hadoop.hive.ql.metadata.Table tbl) throws UnsupportedOperationException, HiveException {
```

Comment:
why do you throw HiveException here?

Reply:
To indicate if table is not partitioned or table is empty :
https://github.com/apache/hive/pull/4346/files#diff-93864ecf035fe51b92185015da842a56837cea89064813de39c278c6f8fed03cR1464

Reply:
👍